pdr_backend/lake/gql_data_factory.py

```diff
@@ -55,28 +59,26 @@ def __init__(self, ppss: PPSS):
 
         contract_list = [f.lower() for f in contract_list]
 
-        # configure all tables that will be recorded onto lake
+        # configure all DB tables <> QGL queries
         self.record_config = {
-            "fetch_functions": {
-                Prediction: fetch_filtered_predictions,
-                Subscription: fetch_filtered_subscriptions,
-                Trueval: fetch_truevals,
-                Payout: fetch_payouts,
-                Slot: fetch_slots,
-            },
+            "fetch_functions": _GQLDF_REGISTERED_LAKE_TABLES,
             "config": {
                 "contract_list": contract_list,
             },
             "gql_tables": [
                 dn.get_lake_table_name()  # type: ignore[attr-defined]
-                for dn in [Prediction, Subscription, Trueval, Payout, Slot]
+                for dn in _GQLDF_REGISTERED_LAKE_TABLES
             ],
         }
 
         TableRegistry().register_tables(
-            [Prediction, Subscription, Trueval, Payout, Slot], self.ppss
+            list(_GQLDF_REGISTERED_LAKE_TABLES.keys()), self.ppss
```

Comment:
I don't think you need the list cast, but I'm not sure.

Reply:
"Given the function register_tables that you've shown, which expects a list of data classes (List[Type[LakeMapper]]), you will indeed need to convert the keys from _GQLDF_REGISTERED_LAKE_TABLES.keys() to a list if those keys represent data class types that are compatible with the register_table method within register_tables."

More simply
- we're not defining the variable and the type
- keys() does not enforce a list
- because we're using type enforce, we should define the type or cast to a type, in order to enforce it